recipes/LibriSpeech/ASR/CTC/train_with_wav2vec_k2.py

```diff
@@ -0,0 +1,656 @@
+#!/usr/bin/env/python3
+"""Recipe for training a wav2vec-based ctc ASR system with librispeech.
+The system employs wav2vec as its encoder. Decoding is performed with
+k2 through the use of a decoding graph and, optionally, a rescoring LM.
+To run this recipe, do the following:
+> python train_with_wav2vec.py hparams/train_{hf,sb}_wav2vec.yaml
+The neural network is trained on CTC likelihood target and character units
+are used as basic recognition tokens.
+
+Authors
+ * Zeyu Zhao 2023
+ * Georgios Karakasidis 2023
+ * Rudolf A Braun 2022
+ * Titouan Parcollet 2022
+ * Sung-Lin Yeh 2021
+ * Ju-Chieh Chou 2020
+ * Mirco Ravanelli 2020
+ * Abdel Heba 2020
+ * Peter Plantinga 2020
+ * Samuele Cornell 2020
+"""
+
+import os
+import sys
+from typing import List, Union
+import torch
+import logging
+import speechbrain as sb
+from speechbrain.utils.distributed import run_on_main, if_main_process
+from hyperpyyaml import load_hyperpyyaml
+from pathlib import Path
+
+from speechbrain.k2_integration.prepare_lang import prepare_lang
+from speechbrain.k2_integration.graph_compiler import CtcTrainingGraphCompiler
+from speechbrain.k2_integration.lexicon import Lexicon, get_lexicon
+
+logger = logging.getLogger(__name__)
+
+
+# Define training procedure
+class ASR(sb.Brain):
+    def compute_forward(self, batch, stage):
+        """Forward computations from the waveform batches to the output probabilities."""
+        batch = batch.to(self.device)
+        wavs, wav_lens = batch.sig
+        wavs, wav_lens = wavs.to(self.device), wav_lens.to(self.device)
+        # Downsample the inputs if specified
```

Comment:
The code layout looks a bit inconsistent thereafter in terms of new lines. Please fix it.  Between two blocks of code: 1 line break. Comments should be attached to the previous line that is targeted.

Reply:
It's exactly the same in `train_with_wav2vec.py`. In any case, I'll add new lines between code and comments.

Reply:
@Adel-Moumen 👀